From 9fd603874caa65791cd74aa88daf61f9f196d5c3 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Fri, 25 Sep 2009 15:07:54 -0400
Subject: [PATCH] Fix gdm_slave_get_timed_login_details

Before it would fail if a NULL username was passed in,
breaking autologin on the multi-stack branch.
---
 daemon/gdm-slave.c |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/daemon/gdm-slave.c b/daemon/gdm-slave.c
index 7bc98c5..94a32ee 100644
--- a/daemon/gdm-slave.c
+++ b/daemon/gdm-slave.c
@@ -881,6 +881,16 @@ gdm_slave_get_timed_login_details (GdmSlave   *slave,
         if (usernamep != NULL) {
                 *usernamep = gdm_slave_parse_enriched_login (slave,
                         username, slave->priv->display_name);
+        } else {
+                g_free (username);
+
+                if (enabledp != NULL) {
+                        *enabledp = enabled;
+                }
+                if (delayp != NULL) {
+                        *delayp = delay;
+                }
+                return TRUE;
         }
         g_free (username);
 
-- 
1.6.4.2

